### PR TITLE
enhance: adding a streaming deserialize reader for binlogs

### PR DESCRIPTION
**Pull request by tedxu**
See #30863

### CODE REVIEW

**Comment by samhuang-z** [comment]
rerun ut

**Comment by samhuang-z** [comment]
rerun ut

**Comment by samhuang-z** [comment]
rerun ut

**Comment by samhuang-z** [comment]
rerun ut

File: internal/storage/serde.go
Comment (sunby) [inline internal/storage/serde.go:1]: maybe we should return an error if the length of splitted blob.Key is less than 2?

File: internal/storage/serde.go [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+
+	id, _ := strconv.ParseInt(parts[len(parts)-1], 0, 10)
+	if len(parts) >= 2 {
+		colId, _ := strconv.ParseInt(parts[len(parts)-2], 0, 10)
+		return colId, id
+	}
+	return id, 0
+}
+
+func newCompositeRecordReader(blobs []*Blob) (*compositeRecordReader, error) {
```
Comment (sunby) [inline internal/storage/serde.go:192]: maybe we should return an error if the length of splitted blob.Key is less than 2?

File: internal/storage/serde.go [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+	return &crr.r
+}
+
+func (crr *compositeRecordReader) Close() {
+	for _, close := range crr.closers {
+		close()
+	}
+}
+
+func getBlobColLog(blob *Blob) (FieldID, UniqueID) {
```
Comment (sunby) [inline internal/storage/serde.go:179]: maybe we should return an error if the length of splitted blob.Key is less than 2?

File: internal/storage/serde.go [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+	return 0
+}
+
+func (r *compositeRecord) Release() {
+	for _, rec := range r.recs {
+		rec.Release()
+	}
+}
+
+func (r *compositeRecord) Schema() *map[FieldID]schemapb.DataType {
```
Comment (sunby) [inline internal/storage/serde.go:71]: better to return map(not the pointer) here

File: internal/storage/serde.go [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+	}
+}
+
+func NewBinlogDeserializeReader(blobs []*Blob, PKfieldID UniqueID) (*DeserializeReader[*Value], error) {
+	reader, err := newCompositeRecordReader(blobs)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewDeserializeReader(reader, func(r Record, v *[]*Value) error {
```
Comment (sunby) [inline internal/storage/serde.go:376]: We can use v as []*Value directly

**Review by congqixia — APPROVED** [review]
/approve

**Pull request merged**